www-apps/redmine/redmine-3.3.3.ebuild

```diff
@@ -0,0 +1,211 @@
+	>=dev-ruby/jquery-rails-3.1.4:3
+	dev-ruby/mime-types:*
+	dev-ruby/mimemagic
+	>=dev-ruby/nokogiri-1.6.8
+	dev-ruby/protected_attributes
+	>=dev-ruby/rails-4.2.5.2:4.2
+	dev-ruby/request_store:1.0.5
+	>=dev-ruby/roadie-rails-1.1.0
+	>=dev-ruby/rbpdf-1.19.2
+	>=dev-ruby/ruby-openid-2.3.0
+	dev-ruby/rubygems
+	fastcgi? ( dev-ruby/fcgi )
+	imagemagick? ( >=dev-ruby/rmagick-2.14.0 )
+	ldap? ( >=dev-ruby/ruby-net-ldap-0.12.0 )
+	markdown? ( >=dev-ruby/redcarpet-3.3.2 )
+	passenger? ( www-apache/passenger )
+	"
+
+#ruby_add_bdepend ">=dev-ruby/rdoc-2.4.2
+#	dev-ruby/yard
+#	test? (
+#		>=dev-ruby/shoulda-3.3.2
+#		>=dev-ruby/mocha-0.13.3
+#		>=dev-ruby/capybara-2.0.0
+#		<dev-ruby/nokogiri-1.6.0
+#	)"
+
+REDMINE_DIR="/var/lib/${PN}"
+
+pkg_setup() {
+	enewgroup redmine
+	enewuser redmine -1 -1 "${REDMINE_DIR}" redmine
+}
+
+all_ruby_prepare() {
+	rm -r log files/delete.me || die
+
+	# bug #406605
+	rm .gitignore .hgignore || die
+
+	rm Gemfile || die
+
+	echo "CONFIG_PROTECT=\"${EPREFIX}${REDMINE_DIR}/config\"" > "${T}/50${PN}"
+	echo "CONFIG_PROTECT_MASK=\"${EPREFIX}${REDMINE_DIR}/config/locales ${EPREFIX}${REDMINE_DIR}/config/settings.yml\"" >> "${T}/50${PN}"
+
+	# remove ldap staff module if disabled to avoid #413779
+	use ldap || rm app/models/auth_source_ldap.rb || die
+
+	# Make it work
+	sed -i -e "1irequire 'request_store'" app/controllers/application_controller.rb || die
+	sed -i -e "18irequire 'action_controller'" -e "19irequire 'action_controller/action_caching'"\
+		app/controllers/welcome_controller.rb || die
+	sed -i -e "4irequire 'action_dispatch/xml_params_parser'" -e "/Bundler/d" config/application.rb || die
+	sed -i -e "18irequire 'protected_attributes'" app/models/custom_field.rb || die
+	sed -i -e "19irequire 'roadie-rails'" app/models/mailer.rb || die
+}
+
+all_ruby_install() {
+	dodoc doc/{CHANGELOG,INSTALL,README_FOR_APP,RUNNING_TESTS,UPGRADING}
+	rm -r doc || die
+	dodoc README.rdoc
+	rm README.rdoc || die
+
+	keepdir /var/log/${PN}
+	dosym /var/log/${PN}/ "${REDMINE_DIR}/log"
+
+	insinto "${REDMINE_DIR}"
+	doins -r .
+	keepdir "${REDMINE_DIR}/files"
+	keepdir "${REDMINE_DIR}/public/plugin_assets"
+
+	fowners -R redmine:redmine \
+		"${REDMINE_DIR}/config" \
+		"${REDMINE_DIR}/files" \
+		"${REDMINE_DIR}/public/plugin_assets" \
+		"${REDMINE_DIR}/tmp" \
+		/var/log/${PN}
+
+	fowners redmine:redmine "${REDMINE_DIR}"
+
+	# protect sensitive data, see bug #406605
+	fperms -R go-rwx \
+		"${REDMINE_DIR}/config" \
+		"${REDMINE_DIR}/files" \
+		"${REDMINE_DIR}/tmp" \
+		/var/log/${PN}
+
+	if use passenger; then
+		has_apache
+		insinto "${APACHE_VHOSTS_CONFDIR}"
+		doins "${FILESDIR}/10_redmine_vhost.conf"
+	else
+		newconfd "${FILESDIR}/${PN}.confd" ${PN}
+		newinitd "${FILESDIR}/${PN}-3.initd" ${PN}
+	fi
+	doenvd "${T}/50${PN}"
+}
+
+pkg_postinst() {
+	einfo
+	if [ -e "${EPREFIX}${REDMINE_DIR}/config/initializers/session_store.rb" -o -e "${EPREFIX}${REDMINE_DIR}/config/initializers/secret_token.rb" ]; then
```

Comment:
Don't ye-ol-sh syntax in bash. Also this ought to use EROOT, i.e.:

    if [[ -e ${EROOT%/}... || -e ${EROOT%/}...